contracts/staking/PoSValidatorManager.sol

```diff
@@ -98,6 +98,14 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
 
     function completeEndValidation(uint32 messageIndex) external {
         Validator memory validator = _completeEndValidation(messageIndex);
+
+        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+        uint256 rewardAmount = $._rewardCalculator.calculateReward(
```

Comment:
Should this reward be calculated above in `initializeEndValidation`, as the TODO there suggests?

I think the reason it would need to be calculated in `intializeEndValidation` is because the uptime proof is included there, which should be used in calculating the rewards. I think we'd either need to store the uptime submitted to be used here, or calculate and store the reward during the call to `initializeEndValidation`.

Reply:
that makes sense to me, thanks for calling it out. I had seen that TODO but had been unsure about it, since the inputs needed for the calculator (start and end times in particular) were readily available at the point where the actual disbursement needs to happen (`completeEndValidation`). But now I see the discussion of `uptime` in the README, and it makes sense to me that the proven uptime is the basis of the rewards, so I re-worked the calculator's inputs to take an uptime rather than a start and an end, and used the proven uptime as that input to the calculation, in https://github.com/ava-labs/teleporter/pull/534/commits/57a519e728da4431a644ebd148f588bad396ad9e